@here/harp-datasource-protocol/lib/TileInfo.ts

```diff
@@ -543,6 +543,12 @@ export class ExtendedTileInfoWriter {
         const storedTechnique = {} as any;
 
         Object.getOwnPropertyNames(technique).forEach(property => {
+            if ((technique as any)[property] instanceof CallExpr) {
```

Comment:
This is wrong, 

1) For technique params dependent on feature, it freezez some random value that is not necessarily important for technique in general, see `AttrScope` enum
2) For technique params that are same for all features, they may be sent to main thread, but in that case you shouldn't evaluate them, but send them after serializing `Expr.toJSON` form but ...

It's not yet supported on main thread side, so it's better to just skip them ATM and think about fixing it in separate issue.

So, in general you should use `makeDecodedTechnique` from `StyleSetEvaluator`. to convert from technique used in decoder to technique variant that may be sent to main thread.

Reply:
Thanks, I fixed this. A block of code also was mowed to prevent unnecessarily iteration on object properties. Do we really need this code:
```
            if (property.startsWith("_")) {
                continue;
            }
 ...
        storedTechnique._index = technique._index;
        storedTechnique._key = technique._key;
        storedTechnique._styleSetIndex = technique._styleSetIndex;

```

Reply:
> Do we really need this code?

@pit-rpg No (speaking with rather  high confidence).

It's leftover from past, when we've kept more private `StyleSetEvaluator` stuff in technique. Now, it contains only what it needs and `_index`, `_key`, and `_styleSetIndex` are actually needed here and there in code.

So i think that plain shallow copy of all props _excluding_ instances of `Expr` is ok both in both places.

Reply:
Done